express/tests/lib/fixedOTP.test.ts

```diff
@@ -0,0 +1,406 @@
+        expect(getFixedOTPCredentialTemporaryPassword("~isValid")).toEqual("");
+        expect(getFixedOTPCredentialTemporaryPassword("isValid")).toEqual("");
+    });
+
+    it("Check getFixedOTPCredentialSecurityCode", () => {
+        expect(getFixedOTPCredentialSecurityCode("~isValid")).toEqual("");
+        expect(getFixedOTPCredentialSecurityCode("isValid")).toEqual("");
+    });
+
+    it("Check getFixedOTPCredentialMobileNumber", () => {
+        expect(getFixedOTPCredentialMobileNumber("~isValid")).toEqual("");
+        expect(getFixedOTPCredentialMobileNumber("isValid")).toEqual("");
+    });
+
+    it("Check getFixedOTPCredentialSMSCode", () => {
+        expect(getFixedOTPCredentialSMSCode("~isValid")).toEqual("");
+        expect(getFixedOTPCredentialSMSCode("isValid")).toEqual("");
+    });
+});
+
+describe("Check when fixedOTP is active", () => {
+    beforeAll(() => {
+        process.env.USE_STUB_OTP = "true";
+        process.env.FIXED_OTP_CREDENTIALS =
+            '[{"Id": "~isValid", "EmailTemplate": "^testuser1.*@unit-testing.com$", "EmailAddress": "testuser1@unit-testing.com", "EmailOTP": "123456", "EmailPassword": "testuser1Password", "Mobile": "07700900000", "SMSCode": "789012"}]';
+
+        fixedOTPInitialise();
+    });
+
+    it("Check verifyMobileUsingOTPCode", () => {
+        let expectException = false;
+        let checkResult = true;
+
+        try {
+            // Valid SMS Code
+            verifyMobileUsingOTPCode("testuser1@unit-testing.com", "789012");
+
+            // Invalid SMS Code
+            expectException = true;
+            verifyMobileUsingOTPCode("testuser1@unit-testing.com", "123456");
+
+            // Expired SMS Code
+            expectException = true;
+            verifyMobileUsingOTPCode("testuser1@unit-testing.com", "999999");
+        } catch (error) {
+            checkResult = false;
+
+            if (error instanceof CodeMismatchException) {
+                checkResult = expectException;
+            }
+        }
+
+        expect(checkResult).toBe(true);
+    });
+
+    it("Check respondToMFAChallengeForFixedOTPCredential", () => {
+        let expectException = false;
+        let checkResult = true;
+
+        try {
+            // Valid MFA Code
+            respondToMFAChallengeForFixedOTPCredential("testuser1@unit-testing.com", "789012");
+
+            // Invalid MFA Code
+            expectException = true;
+            respondToMFAChallengeForFixedOTPCredential("testuser1@unit-testing.com", "123456");
+
+            // Expired MFA Code
+            expectException = true;
+            respondToMFAChallengeForFixedOTPCredential("testuser1@unit-testing.com", "999999");
+        } catch (error) {
+            checkResult = false;
+
+            if (error instanceof CodeMismatchException) {
+                checkResult = expectException;
+            }
+        }
+
+        expect(checkResult).toBe(true);
+    });
```

Comment:
I am not too sure what these tests actually do, not sure if they need to be separate `it` blocks or not? I've commented out all of the calls to `respondToMFAChallengeForFixedOTPCredential` and they still seem to pass

Reply:
Yes because the Exception is only raised when the Method is called with an Invalid Value so if you don't call anything the Tests will pass but that is not a Valid Unit Test!

Reply:
New tests added as suggested.